DataGenerator.py

```diff
@@ -0,0 +1,221 @@
+        """Get embedding and mask of data"""
+        if type is KindOfData.AUDIOSET:
+            n = len(cls.data_audioset)
+            return cls.data_audioset[int(interval[0] * n): int(interval[1] * n)], cls.mask_audioset[int(interval[0] * n): int(interval[1] * n)]
+        if type is KindOfData.LIVE:
+            n = len(cls.data_live)
+            return cls.data_live[int(interval[0] * n): int(interval[1] * n)], cls.mask_live[int(interval[0] * n): int(interval[1] * n)]
+        if type is KindOfData.ALL:
+            n = len(cls.data_audioset)
+            m = len(cls.data_live)
+
+            # cut some piece of data
+            data_a = cls.data_audioset[int(interval[0] * n): int(interval[1] * n)]
+            mask_a = cls.mask_audioset[int(interval[0] * n): int(interval[1] * n)]
+            data_l = cls.data_live[int(interval[0] * m): int(interval[1] * m)]
+            mask_l = cls.mask_live[int(interval[0] * m): int(interval[1] * m)]
+            data = np.vstack([data_a, data_l])
+            mask = np.vstack([mask_a, mask_l])
+
+            # shuffle embeddings and masks of different sets in the same order
+            order = np.random.permutation(mask.shape[0])
+            data = cls.__shuffle(data, order)
+            mask = cls.__shuffle(mask, order)
+            return data, mask
+
+    @classmethod
+    def generate_audioset_sample(cls, n_samples, path_to_data='bal_train.h5',
+                                 path_to_labels='class_labels_indices.csv',
+                                 path_to_genres=None):
+        """Generate n samples using AudioSet"""
+
+        # extract info from audio set
+        music_data, speech_data = cls.__extract_info_audioset(path_to_data, path_to_labels)
+
+        # extract info about music with genres classification
+        if path_to_genres:
+            music_data_new_res = cls.__extract_music_for_genres(path_to_genres)
+            # add data to sample
+            music_data = np.vstack([music_data, music_data_new_res])
+
+        # generate samples
+        cls.data_audioset, cls.mask_audioset = cls.__generate_data_sample(music_data, speech_data, n_samples)
+
+    @classmethod
+    def generate_live_sample(cls, path):
+        """Generate samples from YouTube video, films etc"""
+
+        # read data
+        with open(path, "rb") as handle:
+            data_dict = pickle.load(handle)
+        live_embed = cls.__uint8_to_float32([data_dict['embeddings_list']][0])
+        mask_list = cls.__bool_to_float32([data_dict['mask_list']][0])
+
+        # shuffle embeddings and masks in the same order
+        order = np.random.permutation(mask_list.shape[0])
+        cls.data_live = cls.__shuffle(live_embed, order)
+        cls.mask_live = cls.__shuffle(mask_list, order)
+        # np.zeros((N_samples, seq_len, 1), dtype=np.float32)
+        sh = cls.mask_live.shape
+        cls.mask_live = cls.mask_live.reshape(sh[0], sh[1], 1)
+
+
+    @staticmethod
+    def __shuffle(arr, order):
```

Comment:
Maybe it would be better to add `order = np.random.permutation(mask_list.shape[0])` and related lines to the shuffle function because for now the function doesn't shuffle but sort according to received order

Reply:
Yes, you are right. I fixed it